valora-inc/wallet
# feat: in app swap review screen

### Description

Implements an in app swap review screen using the [Valora-Rest-API](https://github.com/valora-inc/valora-rest-api) / [0x API](https://docs.0x.org/0x-api-swap/api-references).

#### Video
https://user-images.githubusercontent.com/26950305/184464020-87e5d2ea-d2e3-452d-b971-bf9e61cf4caa.mov

### Other changes

Added swap screen and swap review screen to stack navigators.
Added maxSwapSlippagePercentage to remote config values - to allow slippage adjustments from the remote config.

### Tested

Tested locally on iOS on mainnet as 0x API has no support for alfajores.

### How others should test

#### Developers 
prerequisite: set shouldShowSwapMenuInDrawerMenu used in `DrawerNavigator.tsx` to true.
1. Using a funded mainnet wallet navigate to the swap screen
2. Enter an amounts in to or from and selector tokens. (Testing with cREAL is a good idea as there were some issues previously.
3. Once amount is set within your wallet balance tap 'Review Swap'
4. On the review screen check amounts and estimated price modal.
5. Set your device text to large and ensure no text overlap occurs.

#### QA
Once this feature is cut and included in the a release we can enable it for select versions.
prerequisite: a funded mainnet wallet.
1. Navigate to the swap screen
2. Enter an amounts in to or from and selector tokens. (Testing with cREAL is a good idea as there were some issues previously.
3. Once amount is set within your wallet balance tap 'Review Swap'
4. On the review screen check amounts and estimated price modal.
5. Set your device text to large and ensure no text overlap occurs.

### Related issues

- Fixes #2582, RET-318

### Backwards compatibility

Yes
